### PR TITLE
ci: build sample apps

### PR DESCRIPTION
**Pull request by mrehan27**
part of [MBL-421](https://linear.app/customerio/issue/MBL-421/react-native-sample-apps-to-firebase-distribution)

### Changes

- Updated github actions to build sample apps for every pull request created

### CODE REVIEW

File: .github/workflows/build-sample-apps.yml [hunk 1]
```diff
@@ -0,0 +1,234 @@
+name: Build sample apps
+
+on:
+  pull_request: # build sample apps for every commit pushed to an open pull request (including drafts)
+  push:
+    branches: [ main, feature/* ]
```
Comment (ami-aman) [inline .github/workflows/build-sample-apps.yml:6]: Does this mean that the workflow will trigger on any branch that matches this pattern `feature/*` ? 

If yes, then what about other branches that do not follow this pattern ?

Comment (mrehan27) [reply .github/workflows/build-sample-apps.yml:6]: Yes. The workflow will trigger only on Pull Requests, or any commits pushed. to `main` or branches with pattern `feature/*`, similar to workflows in our other repose ([iOS](https://github.com/customerio/customerio-ios/blob/b7d7d4fec2bd4be9f25b3b8937b01c7dc5360e0a/.github/workflows/build-sample-apps.yml#L4-L6), [Android](https://github.com/customerio/customerio-android/blob/a68a9c5c872dad3b9ea4ba951806a82b8268ad2b/.github/workflows/build-sample-apps.yml#L4-L6) and [Flutter](https://github.com/customerio/customerio-flutter/blob/97d92e76af0cd1486b9444be57f3766322c46fb6/.github/workflows/build-sample-apps.yml#L4-L6)) today.

**Review by ami-aman — APPROVED** [review]
Non-blocking comment

File: .github/workflows/build-sample-apps.yml [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+          fi
+
+          if [ "${{ steps.ios_build.outcome }}" != "success" ]; then
+            echo "iOS build failed."
+            BUILD_FAILED=1
+          fi
+
+          if [ "$BUILD_FAILED" == "1" ]; then
+            exit 1
+          fi
```
Comment (Shahroz16) [inline .github/workflows/build-sample-apps.yml:211]:
```suggestion
     - name: Check Build Status
        run: |
          if [ "${{ steps.android_build.outcome }}" != "success" ] || [ "${{ steps.ios_build.outcome }}" != "success" ]; then
            echo "One or more builds failed."
            exit 1
          fi
```

File: .github/workflows/build-sample-apps.yml [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        uses: peter-evans/find-comment@v3
+        id: existing-comment
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: "github-actions[bot]"
+          body-includes: <!-- sample app builds -->
+
+      - name: Create or update comment
+        uses: peter-evans/create-or-update-comment@v4
+        id: create-comment
```
Comment (Shahroz16) [inline .github/workflows/build-sample-apps.yml:36]: "Find Comment" and "Create or update comment" can be combined into a single step ?

**Review by Shahroz16 — APPROVED** [review]
looks good, just suggestions and minor impvorments

File: .github/workflows/build-sample-apps.yml (hunk 2 above)
Comment (mrehan27) [reply .github/workflows/build-sample-apps.yml]: Updated 👍🏻

**Pull request merged**